### PR TITLE
[17.0][MIG] hr_timesheet_task_required

### PR DESCRIPTION
**Pull request by MohamedOsman7**

### CODE REVIEW

**Review by FrancoMaxime — APPROVED** [review]
LGTM: code review

**Comment by MohamedOsman7** [comment]
I encountered a problem which i will describe step by step with screenshots:
1. In the settings, under the project app settings, i did not set the checkmark in the field (Require Tasks on Timesheets):
2024-02-27 14_07_43-Odoo - Settings und 3 weitere Seiten - Geschäftlich – Microsoft​ Edge


2. In the project app, i choosed the project "Research & Development" and opened his settings:
2024-02-27 14_08_23-Odoo - Projects und 3 weitere Seiten - Geschäftlich – Microsoft​ Edge


3. I also did not set the checkmark in the field (Require Tasks on Timesheets):
2024-02-27 14_09_28-Odoo - Research   Development und 3 weitere Seiten - Geschäftlich – Microsoft​ E


4. In the timesheet app, when attempting to create a timesheet entry with the same project **without choosing a task**, i encountered an unexpected error:
2024-02-27 14_10_56-Odoo - My Timesheets und 3 weitere Seiten - Geschäftlich – Microsoft​ Edge


**Apart from the fact that the error occurs at all, where it should not occur, the error message does not match the error specified in the code, which is:**

**(Note: This screenshot is from version 16.0)**

2024-02-27 13_39_08-Odoo - My Timesheets und 7 weitere Seiten - Profil 2 – Microsoft​ Edge

I would assume that something has changed in the new odoo standard in this regard. But i did not find anything **yet**. Any ideas?

File: hr_timesheet_task_required/views/res_config_settings.xml [hunk 1]
```diff
@@ -10,27 +10,15 @@ License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
         <field name="inherit_id" ref="hr_timesheet.res_config_settings_view_form" />
         <field name="arch" type="xml">
             <xpath
-                expr="//div[@data-key='project']//field[@name='module_hr_timesheet']/../../.."
+                expr="//app[@data-string='Project']//block[@name='project_time']"
```
Comment (CRogos) [inline hr_timesheet_task_required/views/res_config_settings.xml]: `expr="//block[@name='project_time']"`

should be sufficient.

File: hr_timesheet_task_required/views/project_project.xml [hunk 2]
```diff
@@ -25,11 +25,11 @@ License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
...
-            <xpath expr="//div[@id='timesheet_settings']" position="after">
+            <xpath expr="//setting[@id='timesheet_settings']" position="after">
                 <div
                     class="col-lg-6 o_setting_box"
                     id="timesheet_task_settings"
-                    attrs="{'invisible': [('allow_timesheets', '=', False)]}"
+                    invisible="allow_timesheets == False"
                 >
                     <div class="o_setting_left_pane">
                         <field name="is_timesheet_task_required" class="oe_inline" />
```
Comment (CRogos) [inline hr_timesheet_task_required/views/project_project.xml]: I think the hole div block should be replaced by this:

```
                    <setting id="timesheet_task_settings" help="Set tasks on timesheet as a mandatory field" invisible="allow_timesheets == False">
                        <field name="is_timesheet_task_required"/>
                    </setting>
```

File: hr_timesheet_task_required/views/account_analytic_line.xml [hunk 3]
```diff
@@ -14,9 +14,8 @@ License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
                 <field name="is_task_required" invisible="1" />
             </field>
             <field name="task_id" position="attributes">
-                <attribute
-                    name="attrs"
-                >{'required':[('is_task_required','=',True)]}</attribute>
+                <attribute name="required">is_task_required == True
```
Comment (CRogos) [inline hr_timesheet_task_required/views/account_analytic_line.xml]: ` <attribute name="required">is_task_required</attribute>`
should be enough.

File: hr_timesheet_task_required/views/account_analytic_line.xml [hunk 4]
```diff
@@ -29,9 +28,8 @@ License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
...
```
Comment (CRogos) [inline hr_timesheet_task_required/views/account_analytic_line.xml]: Same here. remove "== True"

File: hr_timesheet_task_required/views/account_analytic_line.xml [hunk 5]
```diff
@@ -14,9 +14,8 @@ License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
                 <field name="is_task_required" invisible="1" />
```
Comment (CRogos) [inline hr_timesheet_task_required/views/account_analytic_line.xml:14]: invisible needs to be changed to 
`column_invisible="True"`

**Review by CRogos — CHANGES_REQUESTED** [review]
Tested and working, but some code improvements possible.

**Review by CRogos — APPROVED** [review]

**Pull request merged**